nilearn/datasets/func.py

```diff
@@ -2076,35 +2083,44 @@ def fetch_development_fmri(n_subjects=None, reduce_confounds=True,
     # Participants data: ids, demographics, etc
     participants = _fetch_development_fmri_participants(data_dir=data_dir,
                                                         url=None,
-                                                        verbose=verbose)
+                                                        verbose=verbose
+                                                        )[::-1] # so adults come first
+
+    # Download functional and regressors based on participants
+    if adults_or_children not in ['both', 'children', 'adults']:
+        raise ValueError("Wrong value for adults_or_children={0}. "
+                         "Valid arguments are 'adults', 'children' "
+                         "or 'both'".format(adults_or_children))
 
-    max_subjects = len(participants)
+    child_adult = participants['Child_Adult'].tolist()
+    child_count = child_adult.count('child') if adults_or_children != 'adults' else 0
+    adult_count = child_adult.count('adult') if adults_or_children != 'children' else 0
+    max_subjects = adult_count + child_count
+
+    # Check validity of n_subjects 
     if n_subjects is None:
         n_subjects = max_subjects
 
     if (isinstance(n_subjects, numbers.Number) and
             ((n_subjects > max_subjects) or (n_subjects < 1))):
         warnings.warn("Wrong value for n_subjects={0}. The maximum "
-                      "value will be used instead n_subjects={1}"
-                      .format(n_subjects, max_subjects))
+                      "value (for Child_Adult={1}) will be used instead: "
+                      "n_subjects={2}"
+                      .format(n_subjects, adults_or_children, max_subjects))
         n_subjects = max_subjects
 
-    # Download functional and regressors based on participants
-    child_count = participants['Child_Adult'].tolist().count('child')
-    adult_count = participants['Child_Adult'].tolist().count('adult')
-
     # To keep the proportion of children versus adults
-    n_child = np.round(float(n_subjects) / max_subjects * child_count).astype(int)
-    n_adult = np.round(float(n_subjects) / max_subjects * adult_count).astype(int)
+    n_child = (np.round(float(n_subjects) / max_subjects * child_count).astype(int)) if n_subjects>1 else 0
```

Comment:
pep8:  n_subjects > 1

Reply:
This line is also a little too long for PEP8... One (light) suggestion would be:

```python
percent_total = float(n_subjects) / max_subjects
n_child = np.round(percent_total * child_count).astype(int) if n_subjects > 1 else 0
n_adult = np.round(percent_total * adult_count).astype(int) if n_subjects > 1 else 1
```